src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,505 @@
+};
+/**
+ * Returns the internal values subset which is the array that was created from the input JSON.
+ * This array has information for each data point w.r.t shape, colors and on click callback along with
+ * x and y co-ordinates.
+ *
+ * @private
+ * @param {object} target - Object containing the subsets
+ * @returns {Array} List of data point subsets
+ */
+const getDataPointValues = (target) => target.internalValuesSubset;
+/**
+ * Checks the data-set is currently shown in the graph and if the y data-point value is null
+ * If they are then true, false otherwise
+ *
+ * @private
+ * @param {object} shownTargets - graph targets config object
+ * @param {object} value - data point value object
+ * @returns {boolean} true if data point needs to be hidden, false otherwise
+ */
+const shouldHideDataPoints = (shownTargets, value) =>
+    shownTargets.indexOf(value.key) < 0 || value.y === null;
+
+/**
+ * Draws the points with options opted in the input JSON by the consumer for each data set.
+ *  Render the point with appropriate color, x and y co-ordinates, label etc.
+ *  On click content callback function is called.
+ *
+ * @private
+ * @param {object} scale - d3 scale for Graph
+ * @param {object} config - Graph config object derived from input JSON
+ * @param {Array} pointGroupPath - d3 html element of the points group
+ * @param {object} dataTarget - data for the bubble graph
+ * @returns {undefined} - returns nothing
+ */
+const drawBubbles = (scale, config, pointGroupPath, dataTarget) => {
+    // Add a scale for bubble size
+    let bubbleScale;
+    if (!utils.isUndefined(dataTarget.weight)) {
+        bubbleScale = d3.scale
+            .linear()
+            .domain([dataTarget.weight.min, dataTarget.weight.max])
+            .range([
+                constants.DEFAULT_BUBBLE_RADIUS_MIN,
+                constants.DEFAULT_BUBBLE_RADIUS_MAX
+            ]);
+    }
+
+    const renderDataPoint = (path, value, index) => {
+        path.append("g")
+            .classed(styles.point, true)
+            .attr("aria-disabled", !utils.isDefined(value.onClick))
+            .attr("transform", transformPoint(scale)(value))
+            .attr("aria-hidden", (value) =>
+                shouldHideDataPoints(config.shownTargets, value)
+            )
+            .on("click", () =>
+                dataPointActionHandler(
+                    value,
+                    index,
+                    d3
+                        .select(`g[aria-describedby="${value.key}"`)
+                        .selectAll(`.${styles.point}`)[0][index]
+                )
+            )
+            .append("circle")
+            .attr("aria-describedby", value.key)
+            .attr("r", (d) => {
+                const weightAreDefined = utils.isDefined(dataTarget.weight)
+                    ? utils.isDefined(dataTarget.weight.min) &&
+                      utils.isDefined(dataTarget.weight.max)
+                    : false;
+                if (
```

Comment:
Can you split all of these into functions? 
I think this can be much more readable and maitainable that way.

Reply:
Ok, I will make the change